internal/service/enrollmentrequest.go

```diff
@@ -160,8 +159,11 @@ func (h *ServiceHandler) ReadEnrollmentRequest(ctx context.Context, request serv
 // (PUT /api/v1/enrollmentrequests/{name})
 func (h *ServiceHandler) ReplaceEnrollmentRequest(ctx context.Context, request server.ReplaceEnrollmentRequestRequestObject) (server.ReplaceEnrollmentRequestResponseObject, error) {
 	orgId := store.NullOrgId
-	if request.Body.Metadata.Name == nil || request.Name != *request.Body.Metadata.Name {
-		return server.ReplaceEnrollmentRequest400Response{}, nil
+	if request.Body.Metadata.Name == nil {
+		return server.ReplaceEnrollmentRequest400JSONResponse{Message: "resource name not specified in metadata"}, nil
```

Comment:
I'd prefer to have exact path in the error msg like 
```suggestion
		return server.ReplaceEnrollmentRequest400JSONResponse{Message: "metadata.name is not specified"}, nil
```

I believe this aligns with what k8s does. It is especially handy for more complicated paths.